### PR TITLE
Add IBM Power VS: tf data

### PR DESCRIPTION
**Pull request by clnperez**
For more background on IPI on Power VS, refer to the enhancement
proposal here: openshift/enhancements#736

Older discussions on some of the code here can be found in #5224

Signed-off-by: Christy Norman <christy@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by clnperez** [comment]
this needs some rework post https://github.com/openshift/installer/pull/5507 and to stop relying on the custom s3 plugin for the bootstrap ignition config

**Comment by clnperez** [comment]
i'm going to add a few more bits to this so we have fewer PRs to review, b/c of the timing with the tests mentioned in other Power VS PRs.

**Comment by clnperez** [comment]
@patrickdillon @rna-afk  -- I'm taking this back out of WIP and will push new changes to a new PR. The "small thing" I wanted to add here turned out to be a little more like a medium thing.

**Comment by clnperez** [comment]
@rna-afk @patrickdillon @Prashanth684 can you take a look at this as well? Depends on https://github.com/openshift/installer/pull/5615

**Comment by clnperez** [comment]
ping @rna-afk do you know if the list of required/optional tests only shows up after an lgtm?

**Comment by rna-afk** [comment]
@patrickdillon has been working on clearing up the CI tests that are blocking PRs from merging. I'm hoping that's the reason we aren't seeing any required tests right now.

File: data/data/powervs/topology/pi_network.tf [hunk 1]
```diff
@@ -0,0 +1,40 @@
+## Network
+## These are be optional arguments in the install-config (e.g. Platform)
+## so that users may specify them. Have them be "hidden" in that the survey doesn't ask for them
+## unless the OCP leads disagree.
+## And since they're optional, use the count = construct to conditionally create them if the tf
+## vars are unset.
+
+## Note, the following are incomplete placeholders to be tested and reviewed later when the TF
```
[the remark this answers is not in the document]
Comment (clnperez) [reply data/data/powervs/topology/pi_network.tf:8]: no those i probably should have deleted but they're removed in #5780

**Comment by clnperez** [comment]
@mjturek -- can you double-check the descriptions I just added (esp the COS ones for bootstrap)?

File: data/data/powervs/cluster/bootstrap/main.tf [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# TODO(mjturek): network and image data blocks can be in main module
+#                as master and bootstrap will be using the same
+#                network and image. Once we add in master module, make
+#                the move.
```
Comment (mjturek) [inline data/data/powervs/cluster/bootstrap/main.tf:4]: This comment is no longer relevant, we can drop it.

File: data/data/powervs/cluster/bootstrap/variables.tf [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+variable "processors" {
+  type = string
+  description = "The processor count for each node."
+}
+
+variable "ignition" {
+  type = string
+  description = "The ignition file."
```
Comment (mjturek) [inline data/data/powervs/cluster/bootstrap/variables.tf:13]: I think this is okay, but "The contents of the ignition file" may be more accurate.

File: data/data/powervs/cluster/bootstrap/variables.tf [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+variable "cos_instance_location" {
+  type = string
+  description = "The region in which the Cloud Object Store resides. Used for the ignition file."
+}
+
+variable "cos_bucket_location" {
+  type = string
+  description = "The URL of the Cloud Object Store. Used for the igntion file."
```
Comment (mjturek) [inline data/data/powervs/cluster/bootstrap/variables.tf:63]: This is actually used for `region_location` [here](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/cos_bucket#region_location).

So it's not a URL, but a region like `us-south` or `eu-gb`.

**Review by mjturek — CHANGES_REQUESTED** [review]
Comments inline

File: data/data/powervs/cluster/bootstrap/main.tf (hunk 2 above)
Comment (clnperez) [reply data/data/powervs/cluster/bootstrap/main.tf:4]: that's also dropped in #5780

File: data/data/powervs/cluster/bootstrap/variables.tf (hunk 4 above)
Comment (clnperez) [reply data/data/powervs/cluster/bootstrap/variables.tf:63]: how would i differentiate that from `cos_instance_location` above?

Comment (mjturek) [reply data/data/powervs/cluster/bootstrap/variables.tf:63]: There isn't much difference, but `cos_instance_location` can be set to `global` unlike `region_location`. We actually use the `global` setting for `cos_instance_location` so as far as I can tell, we should keep both.

**Comment by clnperez** [comment]
thanks @mjturek!

should be all good now @rna-afk

**Comment by clnperez** [comment]
/retest-required

@patrickdillon these two keep failing for this and #5615. do i need to keep retrying? if a PR gets an approve does mergebot automatically retry for me?

File: data/data/powervs/topology/pi_network.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (clnperez) [reply data/data/powervs/topology/pi_network.tf:8]: @rna-afk i'm going to resolve this an assume it's okay to leave as-is b/c it's taken care of in the other pr. lmk if that's not okay.

[the remark this answers is not in the document]
Comment (patrickdillon) [reply data/data/powervs/topology/pi_network.tf:8]: I would prefer not to merge code to remove it in a later pr/commit & #5780 still needs to be broken into separate commits. Can we get rid of this and the other unnecessary comments?

**Review by patrickdillon — COMMENTED** [review]
This looks very close. But I would prefer to remove the unneeded comments, especially considering one is an entire file.

File: data/data/powervs/topology/pi_network.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (clnperez) [reply data/data/powervs/topology/pi_network.tf]: sure. deleted this whole file and re-pushed

**Comment by clnperez** [comment]
@patrickdillon can you skip those two for this pr as well? nothing seems to be re-running anyway

**Pull request merged**